### PR TITLE
add global autoload directories to search before local ones

### LINKED ISSUE #331: "autoload" feature not working on *NIX
Running Woof! from the build tree does not load any autoload data.

In *NIX there is a general distinction between system data and user data. When the engine gets installed into the system, its autoload data must get installed into a system wide directory (e.g. `/usr/share/woof/`) which is read-only to the casual user. Thus, the casual user needs to be able to supply his own autoload directory, which currently resides under `~/.local/share/woof/autoload/`.

This means, we need an install rule for the autoload data and prepare the autoload code to look in two directories on *NIX systems, i.e. the system autoload directory first and the user owned autoload directory second. The build directory should be out of question for *NIX systems, but we should probably provide a warning to stderr that certain features are going to be missing if the engine hasn't been properly installed on the system.

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #331

### CODE REVIEW

**Comment by rfomin** [comment]
I  tested this on Windows - it works. `cpack` works too.

File: Source/d_main.c [hunk 1]
```diff
@@ -663,27 +663,62 @@ char *D_DoomPrefDir(void)
...
+    return result;
+}
 
-    result = M_StringJoin(autoload_path, DIR_SEPARATOR_S, iwadname, NULL);
+static void PrepareAutoloadPaths (void)
+{
+    int i;
 
-    if (createdir)
+    if (autoload_paths)
```
Comment (rfomin) [inline Source/d_main.c:697]: I cannot understand this check. Why is it needed and is it always not NULL?

Comment (fabiangreffrath) [reply Source/d_main.c:697]: It is always NULL before the function was called and always non-NULL after it has been called once. Indeed, it is obsolete now. I had the code structure a bit differently earlier and this is a remnant of this.

**Review by rfomin — APPROVED** [review]
It works on Windows. I still don't have *NIX system.

**Comment by fabiangreffrath** [comment]
Thanks for the review!

**Pull request merged**